ModuleWork3.1/ModuleWork3.1/Repositories/ContactBookRepository.cs

```diff
@@ -0,0 +1,133 @@
+﻿using System.Collections.Concurrent;
+using System.Text;
+using ModuleWork3._1.Models;
+using ModuleWork3._1.Services;
+
+namespace ModuleWork3._1.Repositories;
+
+internal class ContactBookRepository : IContactBookRepository
+{
+    private readonly string filePath = "contacts.txt";
+
+    internal ConcurrentDictionary<uint, Contact> contacts = new();
+
+    private static SemaphoreSlim fileSemaphore = new(1, 1);
```

Comment:
If your class holds reference to the unmanaged resource OR managed resource that implements IDIsposable, it is a good practice to implement IDIsposable for your class.